prompt for passphase to unlock only during signup

mnaamani opened the pull request:
Only prompt for password when running interactively (during signup)
Otherwise passphrase must be supplied as an argument `--passphrase myPassword`

bwhm commented:
Will review properly later but can add a couple of outstanding items:

* `$ colossus --help`
Does not list `--passphrase`

* Wrong/missing passphrase returns:
`TypeError: Cannot read property 'address' of undefined`
Maybe add some extra info?

* If you don't add a passphrase, the `--passhprase <whatever>` will be ignored.
Ie. not sure if it's even something to fix, as it works.

mnaamani commented:
>  $ colossus --help
> Does not list --passphrase

Added description to help output.

> Wrong/missing passphrase returns:
> TypeError: Cannot read property 'address' of undefined
> Maybe add some extra info?

Fixed, now says:
`Error: Failed to ulock storage provider account`

>     * If you don't add a passphrase, the `--passhprase <whatever>` will be ignored.
>       Ie. not sure if it's even something to fix, as it works.

Yes I think that is acceptable, since it is optional argument after all.

mnaamani commented:
>     * If you don't add a passphrase, the `--passhprase <whatever>` will be ignored.
>       Ie. not sure if it's even something to fix, as it works.

I think I understand more clearly what you meant now. If the actual key-file isn't encrypted, and the supplied passphrase isn't required, its not being communicated that it was ignored. I added a debug message for that case.
